numpy/core/arrayprint.py

```diff
@@ -521,7 +513,10 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
-    if a.size == 0:
+    if a.shape == () and a.dtype.names is None:
```

Comment:
Sorry, looking better I see you explicitly comment on the lack of ability to format scalars. Anyway, the `style=None` option to `array2string` would at least allow that to be remedied in explicit calls.